agent/taskresource/logrouter/logconfig_unix.go

```diff
@@ -0,0 +1,145 @@
+// +build linux
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package logrouter
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	generator "github.com/awslabs/go-config-generator-for-fluentd-and-fluentbit"
+)
+
+const (
+	// LogRouterTypeFluentd is the type of a fluentd log router.
+	LogRouterTypeFluentd = "fluentd"
+
+	// LogRouterTypeFluentbit is the type of a fluentbit log router.
+	LogRouterTypeFluentbit = "fluentbit"
+
+	// socketInputNameFluentd is the name of the socket input plugin for fluentd.
+	socketInputNameFluentd = "unix"
+
+	// socketInputNameFluentbit is the name of the socket input plugin for fluentbit.
+	socketInputNameFluentbit = "forward"
+
+	// socketInputPathOptionFluentd is the key for specifying socket path for fluentd.
+	socketInputPathOptionFluentd = "path"
+
+	// socketInputPathOptionFluentbit is the key for specifying socket path for fluentbit.
+	socketInputPathOptionFluentbit = "unix_path"
+
+	// outputTypeLogOptionKeyFluentd is the key for the log option that specifies output plugin type for fluentd.
+	outputTypeLogOptionKeyFluentd = "@type"
+
+	// outputTypeLogOptionKeyFluentbit is the key for the log option that specifies output plugin type for fluentbit.
+	outputTypeLogOptionKeyFluentbit = "Name"
+
+	// includePatternKey is the key for include pattern.
+	includePatternKey = "include-pattern"
+
+	// excludePatternKey is the key for exclude pattern.
+	excludePatternKey = "exclude-pattern"
+
+	// socketPath is the path for socket file.
+	socketPath = "/var/run/fluent.sock"
+)
+
+// generateConfig generates a FluentConfig object that contains all necessary information to construct
+// a fluentd or fluentbit config file for the log router.
+func (logRouter *LogRouterResource) generateConfig() (generator.FluentConfig, error) {
+	config := generator.New()
+
+	// Specify log stream input, which is a unix socket that will be used for communication between log router container
+	// and other containers.
+	var inputName, inputPathOption string
+	if logRouter.logRouterType == LogRouterTypeFluentd {
+		inputName = socketInputNameFluentd
+		inputPathOption = socketInputPathOptionFluentd
+	} else {
```

Comment:
do we anticipate only these two router types or can there be third in the future?

Reply:
There could theoretically be a third, but realistically for the near future (like the next year or more) there will only be these 2

Reply:
in that case, let's explicitly check if the router type is fluentbit in the `else`, instead of assuming it.

Reply:
i already checked that there can only be two types here: https://github.com/aws/amazon-ecs-agent/pull/2097/commits/c643d889f318f738f9b965aed7a4e6e2d1e9392a#diff-27d3a72427896263cf317001fda17336R258. so i don't think we need to check again..